clients/python-wrapper/lakefs/client.py

```diff
@@ -106,6 +110,70 @@ def version(self) -> str:
         return self._server_conf.version
 
 
+def get_identity_token(session: 'boto3.Session') -> str:
+    """
+   Generate the identity token required for lakeFS authentication from an AWS session.
+
+   This function uses the STS client to generate a presigned URL for the `get_caller_identity` action, extracts the required values from the URL,
+   and creates a base64-encoded JSON object with these values.
+
+   :param session: A boto3 session object with the necessary AWS credentials and region information.
+   :return: A base64-encoded JSON string containing the required authentication information.
+   :raises ValueError: If the session does not have a region name set.
+   """
+
+    if session.region_name is None:
+        raise ValueError("Region name is not set in the session")
+
+    from botocore.client import Config
+    sts_client = session.client('sts', config=Config(signature_version='v4'))
+    presigned_url = sts_client.generate_presigned_url('get_caller_identity', Params={}, ExpiresIn=3600)
+
+    # Parse the URL to extract components
+    parsed_url = urlparse(presigned_url)
+    query_params = parse_qs(parsed_url.query)
+
+    # Extract values from query parameters
+    json_object = {
+        "method": "POST",
+        "host": parsed_url.hostname,
+        "region": session.region_name,
+        "action": query_params['Action'][0],
+        "date": query_params['X-Amz-Date'][0],
+        "expiration_duration": "3600",  # 1 hour in seconds
+        "access_key_id": query_params['X-Amz-Credential'][0].split('/')[0],
+        "signature": query_params['X-Amz-Signature'][0],
+        "signed_headers": query_params['X-Amz-SignedHeaders'],
+        "version": query_params['Version'][0],
+        "algorithm": query_params['X-Amz-Algorithm'][0],
+        "security_token": query_params.get('X-Amz-Security-Token', [None])[0]
+    }
+    json_string = json.dumps(json_object)
+    return base64.b64encode(json_string.encode('utf-8')).decode('utf-8')
+
+
+def from_aws_role(session: 'boto3.Session', ttl_seconds: int = 3600, **kwargs) -> Client:
```

Comment:
Maybe `with_aws_role`?

Reply:
1. It's a client that is generated from the AWS role
2. We already have `from_web_identity`

Reply:
I just looked at `from_web_identity` a second ago and had the same thought. Please ignore 😄